src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -134,15 +135,24 @@ UInt SpatialPooler::getNumInputs() const { return numInputs_; }
 UInt SpatialPooler::getPotentialRadius() const { return potentialRadius_; }
 
 void SpatialPooler::setPotentialRadius(UInt potentialRadius) {
-  NTA_CHECK(potentialRadius < numInputs_);
+  NTA_CHECK(potentialRadius < numInputs_/2);
+  for(const auto dim: inputDimensions_) {
+    if(potentialRadius > (dim/2) +1) {
+      NTA_WARN << "potentialRadius >= one of the dimensions: " << dim;
+    }
+  } 
   potentialRadius_ = potentialRadius;
+  NTA_CHECK((UInt)(potentialPct_ * potentialRadius_) >= 1u) << "SP: at least 1 input synapse must be able to activate from potential pool.";
+  NTA_CHECK(stimulusThreshold_ <= potentialPct_ * potentialRadius) << "Stimulus threshold must be <= than the number of possibly active input synapses per segment.";
```

Comment:
`potentialPct_ * potentialRadius` is not the size of the potential pool, in the case where there are multiple dimensions.  For this you can query the connections class?

Reply:
> is not the size of the potential pool, in the case where there are multiple dimensions 

This is from the time when SP was essentially 1D and any nD inputs/SP were simulated by converting to 1D columnar array. 

I was going to suggest changing `Real potentialRadius` to `vector<Real> potentialRadii` where size potentialRadii == num dimensions of inputs

> For this you can query the connections class? 

no comprende. afaik the Connections doesn't have a concept of potential pools(?) Do you suggest moving this functionality there?